frontend/src/layout/Navbar.tsx

```diff
@@ -1,15 +1,48 @@
 import React, { Component } from 'react';
 import '../index.css';
 import { connect } from 'react-redux';
-
+import { User } from '../actions/types';
+import { logoutUser } from '../actions/user';
+import { Link } from 'react-router-dom';
 interface Props {
-  currentUserToken: string;
+  SessionData?: {
+    user: User;
+    token: string;
+  };
+  logoutUser: Function;
 }
 
 class _Navbar extends Component<Props> {
+  logoutSession = (): void => {
+    if (this.props.SessionData) {
+      this.props.logoutUser(this.props.SessionData.user.id);
+    }
+  };
+
   getLoginState() {
     return (
-      <p className='mr-2'>{this.props.currentUserToken ? 'Logged In' : ''}</p>
+      <div className='mr-2'>
+        {this.props.SessionData ? (
+          <div>
+            <span className='text-lg font-bold mr-4'>{`Hello, ${this.props.SessionData.user.name}!`}</span>
+            <button
+              className='btn btn-sm btn-info mr-5'
+              onClick={this.logoutSession}
+            >
+              Logout
+            </button>
+          </div>
+        ) : (
+          <div>
+            <Link to={'/login'} className='btn btn-sm btn-info mr-5'>
+              Login
+            </Link>
```

Comment:
```suggestion
            <Link to='/login' className='btn btn-sm btn-info mr-5'>
              Login
            </Link>
```
It is okay to use quotes on this one since we don't have dynamic data, please change also similar to this